### PR TITLE
Persistent volume, simplified docker images and avoid localstack

### PR DESCRIPTION
**Pull request by afsalthaj**
* Removed ENV and configs from docker images. This created confusion and bugs
* Updated docker compose with persistent volume for template store, caching compiled templates as well as Redis
* Removed localstack for the following reason:
   * Connection errors when downloading and uploading cached templates. Removed it as we don't want to expose inefficient set up. Things worked before , even if caching upload fails, worker can be launched, which is good.
   * No simple persistence I could find with localstack
   
 The connection error I found when uploading/downloading compiled templates
 
 ```scala
 timestamp":"2024-01-18T07:57:55.896743Z","level":"WARN","message":"Failed to upload compiled template TemplateKey { template_id: TemplateId(cc2cffa9-57d5-4165-a452-0108c954345b), template_version: 0 }: Failed to download template: cc2cffa9-57d5-4165-a452-0108c954345b#0: Could not upload compiled template: dispatch failure: io error: error trying to connect: dns error: failed to lookup address information: Name or service not known: dns error: failed to lookup address information: Name or service not known: failed to lookup address information: Name or service not known (DispatchFailure(DispatchFailure { source: ConnectorError { kind: Io, source: hyper::Error(Connect, ConnectError(\"dns error\", Custom { kind: Uncategorized, error: \"failed to lookup address information: Name or service not known\" })), connection: Unknown } }))","target":"golem_worker_executor_base::services::template"}
 ```

(I was able to connect to localstack s3 from other containers though, questioning the Rust code with custom endpoint URL

```scala
Unpacking less (590-2) ...
Setting up less (590-2) ...
Processing triggers for mailcap (3.70+nmu1) ...
# aws s3 ls --endpoint-url http://localstack:4566
2024-01-18 05:18:44 golem-compiled-components
# %
 ✘ afsalthaj@Afsals-MacBook-Pro  ~/projects/golem-services   persistent_volume ±  docker ps
CONTAINER ID   IMAGE                                              COMMAND                  CREATED         STATUS                   PORTS                                                                                                                             NAMES
```

Persistence of local stack, find info here:

image

### CODE REVIEW

File: golem-worker-executor-base/src/services/compiled_template.rs [hunk 1]
```diff
@@ -47,6 +48,7 @@ pub async fn configured(
                 aws_config::defaults(BehaviorVersion::v2023_11_09()).region(Region::new(region));
 
             let sdk_config = if let Some(endpoint_url) = &config.aws_endpoint_url {
+                info!("The endpoint urls is {}", &endpoint_url);
```
Comment (afsalthaj) [inline golem-worker-executor-base/src/services/compiled_template.rs:51]: This didn't work at all , while I was able to login to worker container and query the s3 buckets in localstack usintg the endpoint URL

The issue is

```scala
timestamp":"2024-01-18T07:57:55.896743Z","level":"WARN","message":"Failed to upload compiled template TemplateKey { template_id: TemplateId(cc2cffa9-57d5-4165-a452-0108c954345b), template_version: 0 }: Failed to download template: cc2cffa9-57d5-4165-a452-0108c954345b#0: Could not upload compiled template: dispatch failure: io error: error trying to connect: dns error: failed to lookup address information: Name or service not known: dns error: failed to lookup address information: Name or service not known: failed to lookup address information: Name or service not known (DispatchFailure(DispatchFailure { source: ConnectorError { kind: Io, source: hyper::Error(Connect, ConnectError(\"dns error\", Custom { kind: Uncategorized, error: \"failed to lookup address information: Name or service not known\" })), connection: Unknown } }))","target":"golem_worker_executor_base::services::template"}
```

while I was able to do

````
Unpacking less (590-2) ...
Setting up less (590-2) ...
Processing triggers for mailcap (3.70+nmu1) ...
# aws s3 ls --endpoint-url http://localstack:4566
2024-01-18 05:18:44 golem-compiled-components
# %
 ✘ afsalthaj@Afsals-MacBook-Pro  ~/projects/golem-services   persistent_volume ±  docker ps
CONTAINER ID   IMAGE                                              COMMAND                  CREATED         STATUS                   PORTS                                                                                                                             NAMES
```

File: docker-compose.yml [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
       - "6379:6379"
 
   golem-shard-manager:
     build:
       context: .
       dockerfile: golem-shard-manager/docker/amd64/Dockerfile
-      args:
-        - REDIS_HOST=redis
+    environment:
+      - REDIS_HOST=redis
```
Comment (justcoon) [inline docker-compose.yml:16]: is this env var needed ? as there is GOLEM__REDIS__HOST

File: docker-compose.yml [hunk 3]
```diff
@@ -42,15 +52,36 @@ services:
...
+      - ENVIRONMENT=local
+      - RUST_LOG=info
+      - WASMTIME_BACKTRACE_DETAILS=1
+      - RUST_BACKTRACE=1
+      - GOLEM__REDIS__PORT=6379
+      - GOLEM__REDIS__HOST=redis
+      - GOLEM__TEMPLATE_SERVICE__CONFIG__HOST=golem-cloud-server-oss
+      - GOLEM__TEMPLATE_SERVICE__CONFIG__PORT=9090
+        # This is a dummy access id, didn't want to change the base golem-config as I am not aware of what's the implication
+      - GOLEM__TEMPLATE_SERVICE__CONFIG__ACCESS_TOKEN="2A354594-7A63-4091-A46B-CC58D379F677"
```
Comment (justcoon) [inline docker-compose.yml:65]: there is not auth in OSS such variable should not be needed

File: docker-compose.yml [hunk 4]
```diff
@@ -42,15 +52,36 @@ services:
...
+      - GOLEM__SHARD_MANAGER_SERVICE__CONFIG__PORT=9001
+      - GOLEM__SHARD_MANAGER_SERVICE__CONFIG__RETRIES__MAX_ATTEMPTS=5
+      - GOLEM__SHARD_MANAGER_SERVICE__CONFIG__RETRIES__MIN_DELAY="100ms"
+      - GOLEM__SHARD_MANAGER_SERVICE__CONFIG__RETRIES__MAX_DELAY="2s"
+      - GOLEM__SHARD_MANAGER_SERVICE__CONFIG__RETRIES__MULTIPLIER=2
+      - GOLEM__BLOB_STORE_SERVICE__CONFIG__REGION=us-east-1
+      - GOLEM__COMPILED_TEMPLATE_SERVICE__CONFIG__ROOT="data/templates"
+      - GOLEM__COMPILED_TEMPLATE_SERVICE__TYPE="Local"
+      - GOLEM__SHARD_MANAGER_SERVICE__TYPE="Grpc"
+      - GOLEM__ENABLE_JSON_LOG=true
```
Comment (justcoon) [inline docker-compose.yml:78]: i think in this case it is better to not use json logging

File: docker-compose.yml [hunk 5]
```diff
@@ -1,25 +1,26 @@
...
       dockerfile: golem-shard-manager/docker/amd64/Dockerfile
-      args:
-        - REDIS_HOST=redis
+    environment:
+      - REDIS_HOST=redis
+      - WASMTIME_BACKTRACE_DETAILS=1
+      - RUST_BACKTRACE=1
+      - GOLEM__REDIS__HOST=redis
+      - GOLEM__REDIS__PORT=6379
+      - GOLEM__ENABLE_JSON_LOG=true
```
Comment (justcoon) [inline docker-compose.yml:21]: i think in this case it is better to not use json logging

**Comment by afsalthaj** [comment]
I am merging this

**Pull request merged**